modules/host/upload_test.go

```diff
@@ -0,0 +1,202 @@
+package host
+
+import (
+	"errors"
+	"io/ioutil"
+	"path/filepath"
+	"testing"
+	"time"
+
+	"github.com/NebulousLabs/Sia/crypto"
+	"github.com/NebulousLabs/Sia/modules"
+	"github.com/NebulousLabs/Sia/types"
+)
+
+const (
+	testUploadDuration = 20 // Duration in blocks of a standard upload during testing.
+)
+
+// uploadTestFile uploads a file to the host from the tester's renter.
+func (ht *hostTester) uploadFile(name string) error {
+	// Check that renting is initialized properly.
+	err := ht.initRenting()
+	if err != nil {
+		return err
+	}
+
+	// Create a file to upload to the host.
+	filepath := filepath.Join(ht.persistDir, name+".testfile")
+	datasize := uint64(1024)
+	data, err := crypto.RandBytes(int(datasize))
+	if err != nil {
+		return err
+	}
+	err = ioutil.WriteFile(filepath, data, 0600)
+	if err != nil {
+		return err
+	}
+
+	// Have the renter upload to the host.
+	fup := modules.FileUploadParams{
+		Filename:    filepath,
+		Nickname:    name,
+		Duration:    testUploadDuration,
+		Renew:       false,
+		ErasureCode: nil,
+		PieceSize:   0,
+	}
+	err = ht.renter.Upload(fup)
+	if err != nil {
+		return err
+	}
+
+	// Wait until the upload has finished.
+	for i := 0; i < 100; i++ {
+		time.Sleep(time.Millisecond * 100)
+
+		// Asynchronous processes in the host access obligations by id.
+		if func() bool {
+			ht.host.mu.Lock()
+			defer ht.host.mu.Unlock()
+
+			for _, ob := range ht.host.obligationsByID {
+				if ob.fileSize() >= datasize {
+					return true
+				}
+			}
+			return false
+		}() {
+			break
+		}
+	}
+
+	// The rest of the upload can be performed under lock.
+	ht.host.mu.Lock()
+	defer ht.host.mu.Unlock()
+
+	if len(ht.host.obligationsByID) != 1 {
+		return errors.New("expecting a single obligation")
+	}
+	for _, ob := range ht.host.obligationsByID {
+		if ob.fileSize() >= datasize {
+			return nil
+		}
+	}
+	return errors.New("ht.uploadFile: upload failed")
+}
+
+// TestRPCUPload attempts to upload a file to the host, adding coverage to the
+// upload function.
```

Comment:
`UPload` -> `Upload` -- and is it necessary to specify that a test adds coverage? I think my test docstrings are usually of the form "TestFoo tests the foo function by barring the baz."